monai/networks/nets/restormer.py

```diff
@@ -0,0 +1,314 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import torch
+import torch.nn as nn
+
+from monai.networks.blocks.cablock import CABlock, FeedForward
+from monai.networks.blocks.convolutions import Convolution
+from monai.networks.blocks.downsample import DownSample, DownsampleMode
+from monai.networks.blocks.upsample import UpSample, UpsampleMode
+from monai.networks.layers.factories import Norm
+
+
+class MDTATransformerBlock(nn.Module):
+    """Basic transformer unit combining MDTA and GDFN with skip connections.
+    Unlike standard transformers that use LayerNorm, this block uses Instance Norm
+    for better adaptation to image restoration tasks."""
+
+    def __init__(
+        self,
+        spatial_dims: int,
+        dim: int,
+        num_heads: int,
+        ffn_expansion_factor: float,
+        bias: bool,
+        LayerNorm_type: str,
+        flash_attention: bool = False,
+    ):
+        super().__init__()
+        use_bias = LayerNorm_type != "BiasFree"
+        self.norm1 = Norm[Norm.INSTANCE, spatial_dims](dim, affine=use_bias)
+        self.attn = CABlock(spatial_dims, dim, num_heads, bias, flash_attention)
+        self.norm2 = Norm[Norm.INSTANCE, spatial_dims](dim, affine=use_bias)
+        self.ffn = FeedForward(spatial_dims, dim, ffn_expansion_factor, bias)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x + self.attn(self.norm1(x))
+        x = x + self.ffn(self.norm2(x))
+        return x
+
+
+class OverlapPatchEmbed(nn.Module):
+    """Initial feature extraction using overlapped convolutions.
+    Unlike standard patch embeddings that use non-overlapping patches,
+    this approach maintains spatial continuity through 3x3 convolutions."""
+
+    def __init__(self, spatial_dims: int, in_c: int = 3, embed_dim: int = 48, bias: bool = False):
+        super().__init__()
+        self.proj = Convolution(
+            spatial_dims=spatial_dims,
+            in_channels=in_c,
+            out_channels=embed_dim,
+            kernel_size=3,
+            strides=1,
+            padding=1,
+            bias=bias,
+            conv_only=True,
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.proj(x)
```

Comment:
```suggestion
class OverlapPatchEmbed(Convolution):
    """
    Initial feature extraction using overlapped convolutions. Unlike standard patch embeddings 
    that use non-overlapping patches, this approach maintains spatial continuity through 3x3 convolutions.
    """

    def __init__(self, spatial_dims: int, in_c: int = 3, embed_dim: int = 48, bias: bool = False):
        super().__init__(
            spatial_dims=spatial_dims,
            in_channels=in_c,
            out_channels=embed_dim,
            kernel_size=3,
            strides=1,
            padding=1,
            bias=bias,
            conv_only=True,
        )
```
Would it work to inherit directly from `Convolution`?